Source/Css/ExCssQuery.cs

```diff
@@ -0,0 +1,286 @@
+        {
+            var func = selector switch
+            {
+                AllSelector allSelector => ops.Universal(),
+                AttrAvailableSelector attrAvailableSelector => ops.AttributeExists(attrAvailableSelector.Attribute),
+                AttrBeginsSelector attrBeginsSelector => ops.AttributePrefixMatch(attrBeginsSelector.Attribute, attrBeginsSelector.Value),
+                AttrContainsSelector attrContainsSelector => ops.AttributeSubstring(attrContainsSelector.Attribute, attrContainsSelector.Value),
+                AttrEndsSelector attrEndsSelector => ops.AttributeSuffixMatch(attrEndsSelector.Attribute, attrEndsSelector.Value),
+                AttrHyphenSelector attrHyphenSelector => ops.AttributeDashMatch(attrHyphenSelector.Attribute, attrHyphenSelector.Value),
+                AttrListSelector attrListSelector => ops.AttributeIncludes(attrListSelector.Attribute, attrListSelector.Value),
+                AttrMatchSelector attrMatchSelector => ops.AttributeExact(attrMatchSelector.Attribute, attrMatchSelector.Value),
+                AttrNotMatchSelector attrNotMatchSelector => ops.AttributeNotMatch(attrNotMatchSelector.Attribute, attrNotMatchSelector.Value), // TODO:,
+                ClassSelector classSelector => ops.Class(classSelector.Class),
+                ComplexSelector complexSelector =>  GetFunc(complexSelector, ops, inFunc),
+                CompoundSelector compoundSelector => GetFunc(compoundSelector, ops, inFunc),
+                FirstChildSelector firstChildSelector => GetFunc(firstChildSelector, ops),
+                LastChildSelector lastChildSelector => GetFunc(lastChildSelector, ops),
+                FirstColumnSelector firstColumnSelector => throw new NotImplementedException(), // TODO:,
+                LastColumnSelector firstColumnSelector => throw new NotImplementedException(), // TODO:,
+                FirstTypeSelector firstTypeSelector => throw new NotImplementedException(), // TODO:,
+                LastTypeSelector firstTypeSelector => throw new NotImplementedException(), // TODO:,
+                ChildSelector childSelector => ops.Child(),
+                ListSelector listSelector => GetFunc(listSelector, ops, inFunc), // TODO:,
+                NamespaceSelector namespaceSelector => throw new NotImplementedException(), // TODO:,
+                PageSelector pageSelector => throw new NotImplementedException(), // TODO:,
+                PseudoClassSelector pseudoClassSelector => GetFunc(pseudoClassSelector, ops, inFunc),
+                PseudoElementSelector pseudoElementSelector => throw new NotImplementedException(), // TODO:,
+                TypeSelector typeSelector => ops.Type(typeSelector.Name),
+                UnknownSelector unknownSelector => throw new NotImplementedException(), // TODO:,
+                IdSelector idSelector => ops.Id(idSelector.Id),
+                _ => throw new NotImplementedException(), // TODO:,
```

Comment:
TODO exists.

Reply:
Removed the todos and add one more case that is not implemented. For the not typed selectors I have to look into excss what kind of selectors this is.